code/modules/power/lighting/light.dm

```diff
@@ -241,13 +238,16 @@
 		var/delay = rand(BROKEN_SPARKS_MIN, BROKEN_SPARKS_MAX)
 		addtimer(CALLBACK(src, .proc/broken_sparks), delay, TIMER_UNIQUE | TIMER_NO_HASH_WAIT)
 
-/obj/machinery/light/process()
-	if (!cell)
+/obj/machinery/light/process(delta_time)
+	if(!cell && !reagents)
 		return PROCESS_KILL
 	if(has_power())
-		if (cell.charge == cell.maxcharge)
+		if (cell.charge == cell.maxcharge && !reagents)
```

Comment:
Why are we checking for !reagents twice?

Reply:
Originally the only circumstances in which process() would fire was to affect the cell charge, or to handle low power mode. I tacked on the first check to prevent PROCESS_KILL from firing early if the emergency power cell wasn't present, but the light was still running from station power. Let me find a good way to rearrange this.